frontend/packages/schema-editor/src/components/SchemaTree/SchemaNode/ActionButtons/ActionButtons.tsx

```diff
@@ -49,12 +49,27 @@ export const ActionButtons = ({ pointer, className }: ActionButtonsProps) => {
 
 const useDeleteNode = (pointer: string, savableModel: SavableSchemaModel) => {
   const { t } = useTranslation();
-  const { setSelectedNodePointer } = useSchemaEditorAppContext();
+  const { setSelectedNodePointer, setSelectedTypePointer, selectedTypePointer } =
+    useSchemaEditorAppContext();
 
   return useCallback(() => {
     if (confirm(t('schema_editor.datamodel_field_deletion_text'))) {
       setSelectedNodePointer(null);
+
+      // Close the types panel when deleting the selected type,
+      // unless it's only a child node being deleted
+      if (pointer === selectedTypePointer) {
+        setSelectedTypePointer(null);
+      }
```

Comment:
Her kan vi godt sette `pointer === selectedTypePointer` inn i en variabel for å gjøre koden mer selvforklarende og slippe å måtte bruke kommentarer:
```suggestion
      const pointsToSelectedType = pointer === selectedTypePointer;
      if (pointsToSelectedType) setSelectedTypePointer(null);
```

Reply:
Det er en mye bedre løsning ja :)